futures-util/src/stream/mod.rs

```diff
@@ -599,6 +601,30 @@ pub trait StreamExt: Stream {
         for_each::new(self, f)
     }
 
+    /// Runs this stream to completion, executing the provided closure for each
+    /// element on the stream. This is similar to `for_each` but may begin
+    /// processing an element while previous elements are still being processed.
+    ///
+    /// When this stream successfully resolves to an item, the closure will be
+    /// called to produce a future. That future will then be added to
+    /// the set of futures to resolve.
+    ///
+    /// The returned value is a `Future` where the `Item` type is `()` and
```

Comment:
The docs don't seem to match, since you're actually returning back the stream.

Reply:
oh good catch! that's wrong on for_each, too

Reply:
Fixed.